### PR TITLE
Completed 2 way communication for slack

### PR DESCRIPTION
**Pull request by NarayanaSabari**
Xmessage for 2 way communication on Slack is created, here is the raised Issue : https://github.com/samagra-comms/adapter/issues/60

### CODE REVIEW

**Comment by NarayanaSabari** [comment]
@CodiumAI-Agent /review

File: packages/adapters/slack/src/lib/slack.bot.config.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export type SlackBotProviderConfig = {
+  botToken: string
+};
```
Comment (chinmoy12c) [inline packages/adapters/slack/src/lib/slack.bot.config.ts:3]: nit: add newline

File: packages/adapters/slack/src/lib/slack.provider.spec.ts [hunk 2]
```diff
@@ -21,4 +21,4 @@ test('should trigger Slack correctly', async () => {
     webhookUrl: 'webhookUrl',
     content: 'chat message',
   });
-});
+});
```
Comment (chinmoy12c) [inline packages/adapters/slack/src/lib/slack.provider.spec.ts:24]: nit: add newline

File: packages/adapters/slack/src/lib/slack.provider.ts [hunk 3]
```diff
@@ -1,24 +1,77 @@
...
+    const xmessage: XMessage = {
+      to: {
+        userID: "admin",
+        bot: true,
+      },
+      from: {
+        userID: msg.user,
+        bot: false,
+      },
+      channelURI: "Bot",
```
Comment (chinmoy12c) [inline packages/adapters/slack/src/lib/slack.provider.ts:36]:
```suggestion
      channelURI: "Slack",
```

File: packages/adapters/slack/src/lib/types.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export type SlackUpdateMessage = {
+date: number,
+ type: string; // The type of event, e.g., "message"
+ user: string; // The user ID of the sender
+ text: string; // The text of the message
+ ts: string; // The timestamp of the message
+ channel: string; // The ID of the channel where the message was sent
+};
```
Comment (chinmoy12c) [inline packages/adapters/slack/src/lib/types.ts:8]: nit: add newline

File: packages/adapters/slack/src/lib/slack.provider.spec.ts [hunk 5]
```diff
@@ -21,4 +21,4 @@ test('should trigger Slack correctly', async () => {
     webhookUrl: 'webhookUrl',
     content: 'chat message',
   });
```
Comment (chinmoy12c) [inline packages/adapters/slack/src/lib/slack.provider.spec.ts:23]: Not sure if this test is correct. The `sendMessage` method expects `channel` to be present, but this test checks for `webhookUrl` and `chat message`.

File: packages/adapters/slack/src/lib/slack.provider.ts [hunk 6]
```diff
@@ -1,24 +1,77 @@
...
-  async sendMessage(data: IChatOptions): Promise<ISendMessageSuccessResponse> {
-    const response = await this.axiosInstance.post(data.webhookUrl, {
-      text: data.content,
-    });
+  constructor(private config: SlackBotProviderConfig) {}
 
-    return {
-      id: response.headers['x-slack-req-id'],
-      date: new Date().toISOString(),
+  async convertMessageToXMsg(msg: SlackUpdateMessage): Promise<XMessage> {
```
Comment (chinmoy12c) [inline packages/adapters/slack/src/lib/slack.provider.ts:20]: can we have a test for this function as well?

**Review by chinmoy12c — CHANGES_REQUESTED** [review]

**Comment by chinmoy12c** [comment]
Hey @NarayanaSabari you would also need to add this adapter to the factory. Refer to the documentation [here](https://docs-git-docs-dynamic-deployer-pravrishtis-projects.vercel.app/developer_guide/guides/creating_new_adapter#3-adding-the-adapter-to-factory) for details.

**Comment by NarayanaSabari** [comment]
Hey @chinmoy12c, Due to some bug in GitHub, my latest commit is not showing in this PR, so I am closing this pr and raising a new PR.

**Comment by NarayanaSabari** [comment]
closing...

**Comment by NarayanaSabari** [comment]
This is the updated PR : https://github.com/BharatSahAIyak/packages/pull/44

**Pull request closed without merging**